gamemodes/husklesph/gamemode/sv_awards.lua

```diff
@@ -0,0 +1,133 @@
+-- last prop death award
+local function lastPropStandingCalcuation()
+	if GAMEMODE.LastRoundResult == 2 then
+		return GAMEMODE.LastPropDeath
+	end
+
+	return nil
+end
+
+
+-- get prop with least movement
+local function leastMovementCalcuation()
+	local minPly
+
+	for _, ply in ipairs(GAMEMODE:GetPlayingPlayers()) do
+		if ply:Team() != 3 then continue end
+
+		if !minPly || ply.PropMovement < minPly.PropMovement then
+			minPly = ply
+		end
+	end
+
+	return minPly
+end
+
+
+-- get prop with most taunts
+local function mostTauntsCalcuation()
+	local maxPly
+
+	for _, ply in ipairs(GAMEMODE:GetPlayingPlayers()) do
+		if ply:Team() != 3 then continue end
+
+		if !maxPly || ply.TauntAmount > maxPly.TauntAmount then
+			maxPly = ply
+		end
+	end
+
+	return (maxPly && maxPly.TauntAmount > 0) && maxPly || nil
```

Comment:
These expressions (this func and others) seem weird. Why do they check `maxPly` multiple times?

Reply:
On the first iteration of the loop maxPly will be nil. In that scenario we just want to set maxPly to ply irregardless of the TauntAmount.
I did experiment with a version that avoided this issue but it just introduced more issues. They were fixable but also ultimately outside of the scope of this PR (but might still be a good idea to implement in the future).

The return value also checking for nil is to address a couple scenarios:
1. Loop doesn't run (possible if there are no players on Props and Hunters team)
2. There are no players on the particular team (possible if it's 1v1 and a player disconnects)

Reply:
I misread the line, which didn't help. This is using short circuiting to produce a value, which I'm not sure I like as it's not as readable IMO. What's wrong with:

```lua
if maxPly && maxPly.TauntAmount then return maxPly end
return nil
```

Rather than relying on a big expression where you have to think about how it's going to get processed?

Reply:
I agree, that's more readable. I'll fix things up to use that style.